lib/Entity/DataSet.php

```diff
@@ -379,10 +379,14 @@ public function getData($filterBy = [], $options = [])
         // Get the Latitude and Longitude ( might be used in a formula )
         if ($displayId == 0) {
             $displayGeoLocation = "GEOMFROMTEXT('POINT(" . $this->config->GetSetting('DEFAULT_LAT') . " " . $this->config->GetSetting('DEFAULT_LONG') . ")')";
+            //Show DataSet-Row on Preview if Filter-KeyField empty
+            $displayName = '';
         }
         else {
             $displayGeoLocation = '(SELECT GeoLocation FROM `display` WHERE DisplayID = :displayId)';
             $params['displayId'] = $displayId;
+            //Get displayName from Database
+            $displayName = $this->getStore()->select('Select display from display where displayid = '.$displayId ,[])[0]['display'];
```

Comment:
You'd be better off using the DisplayFactory for this and handling the situation where a display doesn't exist (this would also allow us to extend with other properties if we chose to.

Something like:

```
try {
  $display = $this->displayFactory->getById($displayId);
  $displayName = $display->display;
} catch (NotFoundException $notFoundException) {
  // set them to empty
  $displayName = '';
}
```